flow/connectors/clickhouse/normalize.go

```diff
@@ -610,30 +612,30 @@ func (c *ClickHouseConnector) NormalizeRecords(
 		return model.NormalizeResponse{}, err
 	}
 
-	if err := c.UpdateNormalizeBatchID(ctx, req.FlowJobName, endBatchID); err != nil {
+	if err := c.UpdateNormalizeBatchID(ctx, req.FlowJobName, endBatchIdInclusive); err != nil {
 		c.logger.Error("[clickhouse] error while updating normalize batch id",
-			slog.Int64("batchID", endBatchID), slog.Any("error", err))
+			slog.Int64("batchID", endBatchIdInclusive), slog.Any("error", err))
 		return model.NormalizeResponse{}, err
 	}
 
 	return model.NormalizeResponse{
-		StartBatchID: normBatchID + 1,
-		EndBatchID:   endBatchID,
+		StartBatchID: startBatchIdInclusive,
```

Comment:
I kind of like the two/three-pointer model more than introducing new concepts to keep track of. Is it the opposite for you?

Reply:
agree with this, kept this model